internal/promotion/runner/builtin/git_pr_opener.go

```diff
@@ -85,6 +89,7 @@ func (g *gitPROpener) run(
 			Status: kargoapi.PromotionPhaseSucceeded,
 			Output: map[string]any{
 				stateKeyPRNumber: prNumber,
+				stateKeyRepoURL:  cfg.RepoURL,
```

Comment:
Now that the output is growing, we could probably be more future-proof by doing this:

```
	if prNumber != -1 {
		// We can don't need to check for the existence of prior output from this
		// step because we'd not have gotten to here if it didn't exist.
		output, _ := stepCtx.SharedState.Get(stepCtx.Alias)
		return promotion.StepResult{
			Status: kargoapi.PromotionPhaseSucceeded,
			// Don't need to validate the type assertion on the output here because we
			// we'd not have gotten to here if it wasn't a map[string]any.
			Output: output.(map[string]any), //nolint: forcetypeassert
		}, nil
	}
```

Reply:
Also note that this short-circuiting logic is probably not required anymore.

GitHub doesn't let me comment on lines too far above this change, but would you please modify the comment about short-circuiting on (new) line 80 like so:

```
	// Short-circuit if shared state has output from a previous execution of this
	// step that contains a PR number.
	//
	// TODO(krancour): This short-circuiting logic may have outlived its
	// usefulness now that we keep track of the current step.
```